pkg/detector/library/detect.go

```diff
@@ -1,39 +1,46 @@
 package library
 
 import (
+	"context"
+
 	"golang.org/x/xerrors"
 
 	ftypes "github.com/aquasecurity/trivy/pkg/fanal/types"
+	"github.com/aquasecurity/trivy/pkg/log"
 	"github.com/aquasecurity/trivy/pkg/types"
 )
 
-// Detect scans and returns vulnerabilities of library
-func Detect(libType ftypes.LangType, pkgs []ftypes.Package) ([]types.DetectedVulnerability, error) {
+// Detect scans language-specific packages and returns vulnerabilities.
+func Detect(ctx context.Context, libType ftypes.LangType, pkgs []ftypes.Package) ([]types.DetectedVulnerability, error) {
 	driver, ok := NewDriver(libType)
 	if !ok {
 		return nil, nil
 	}
 
-	vulns, err := detect(driver, pkgs)
+	vulns, err := detect(ctx, driver, pkgs)
 	if err != nil {
 		return nil, xerrors.Errorf("failed to scan %s vulnerabilities: %w", driver.Type(), err)
 	}
 
 	return vulns, nil
 }
 
-func detect(driver Driver, libs []ftypes.Package) ([]types.DetectedVulnerability, error) {
+func detect(ctx context.Context, driver Driver, pkgs []ftypes.Package) ([]types.DetectedVulnerability, error) {
 	var vulnerabilities []types.DetectedVulnerability
-	for _, lib := range libs {
-		vulns, err := driver.DetectVulnerabilities(lib.ID, lib.Name, lib.Version)
+	for _, pkg := range pkgs {
+		if pkg.Version == "" {
+			log.DebugContext(ctx, "Skipping package without version", log.String("name", pkg.Name))
```

Comment:
In order not to confuse users (we do not skip the package, we simply do not look for vulnerabilities for this package)
```suggestion
			log.DebugContext(ctx, "Skip vulnerability detection for package without version", log.String("name", pkg.Name))
```

Reply:
This log message is shown after `[kubernetes] Detecting vulnerabilities`. Isn't it enough?

```
2024-04-23T13:47:40+04:00       INFO    [kubernetes] Detecting vulnerabilities...
2024-04-23T13:47:40+04:00       DEBUG   [kubernetes] Scanning packages from the file    file_path=""
2024-04-23T13:47:40+04:00       DEBUG   [kubernetes] Skipping package without version   name="cilium"
2024-04-23T13:47:40+04:00       DEBUG   [kubernetes] Skipping package without version   name="hubble-relay"
2024-04-23T13:47:40+04:00       DEBUG   [kubernetes] Skipping package without version   name="hubble-ui"
2024-04-23T13:47:40+04:00       DEBUG   [kubernetes] Skipping package without version   name="k8s.io/kube-proxy"
2024-04-23T13:47:40+04:00       DEBUG   [kubernetes] Skipping package without version   name="kube-dns"
```

Reply:
I'm concerned that users may not see this log message.
But okay. We can always update this if users have questions about it.

Reply:
You're right. People don't see log messages carefully 😄 Updated https://github.com/aquasecurity/trivy/pull/6542/commits/d1116c13aa930330024092bd284a0bc0b074f335